core/src/main/java/novoda/lib/sqliteprovider/util/DatabaseStructure.java

```diff
@@ -0,0 +1,226 @@
+package novoda.lib.sqliteprovider.util;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import novoda.lib.sqliteprovider.sqlite.IDatabaseMetaInfo.SQLiteType;
+
+public class DatabaseStructure {
+
+    private static final String SELECT_TABLES_NAME = "SELECT name FROM sqlite_master WHERE type='table';";
+    private static final String PRAGMA_TABLE_INFO = "PRAGMA table_info('%1$s');";
+    private static final String PRGAMA_INDEX_LIST = "PRAGMA index_list('%1$s');";
+    private static final String PRGAMA_INDEX_INFO = "PRAGMA index_info('%1$s');";
+    private static final String COLUMN_PRIMARY_KEY_INDEX = "pk";
+    private static final String COLUMN_NAME = "name";
+    private static final String COLUMN_TYPE = "type";
+
+    private static final List<String> DEFAULT_TABLES = Collections.singletonList("android_metadata");
+
+    private final SQLiteDatabase database;
+
+    private List<String> createdTables;
```

Comment:
❌  this should really be a local variable in `tables()` and passed to collaborators, it has no reason to be a field. They're not status, and on top of that, they're mutable and can cause bugs down the line if they're used in a different way (e.g., reused without clearing them first). Same for the next two fields, they should be local variables.